site/src/server.rs

```diff
@@ -627,17 +633,28 @@ async fn handle_fs_path(req: &Request, path: &str) -> Option<http::Response<hype
         | "/detailed-query.html"
         | "/help.html"
         | "/status.html" => resolve_template(relative_path).await,
-        _ => TEMPLATES.get_static_asset(relative_path)?,
+        _ => {
+            let (data, compressed) = TEMPLATES.get_static_asset(relative_path, allow_compression);
+            if compressed {
+                path_extension = Some(std::ffi::OsStr::new("br"));
+            }
+            data?
+        }
     };
 
-    let p = Path::new(&path);
-    match p.extension().and_then(|x| x.to_str()) {
+    match path_extension.and_then(|x| x.to_str()) {
         Some("html") => response = response.header_typed(ContentType::html()),
         Some("png") => response = response.header_typed(ContentType::png()),
         Some("json") => response = response.header_typed(ContentType::json()),
         Some("svg") => response = response.header("Content-Type", "image/svg+xml"),
         Some("css") => response = response.header("Content-Type", "text/css"),
         Some("js") => response = response.header("Content-Type", "application/javascript"),
+        Some("br") => {
+            response = response.header(
```

Comment:
Let's set this header near line 638, instead of changing the extension. When we compress the data, we should still keep e.g. the `Content-Type` encoding for the original extension.

Reply:
I was concerned that the original requested path might be for a '.br' file. I can remove this if you don't think that would happen.

Reply:
There's not really any reason to support that, I think :)